client/spring-zeebe/src/main/java/io/camunda/zeebe/spring/client/jobhandling/JobHandlerInvokingSpringBeans.java

```diff
@@ -73,7 +73,10 @@ public JobHandlerInvokingSpringBeans(ZeebeWorkerValue workerValue,
   public void handle(JobClient jobClient, ActivatedJob job) throws Exception {
     if (outboundConnectorConfiguration!=null) {
       LOG.trace("Handle {} and execute connector {}", job, outboundConnectorConfiguration);
-      new SpringConnectorJobHandler(outboundConnectorConfiguration, secretProvider, commandExceptionHandlingStrategy, metricsRecorder).handle(jobClient, job);
+      metricsRecorder.executeWithTimer(job.getType(), () -> {
```

Comment:
Two questions:
1. Why do we limit the time recording to connectors - I think this would be valuable to any Job Worker - or WDYT?
2. If we limit it to connectors, why not do it within the SpringConnectorJobHander as this already implements the counter - and thus we would rip apart metrics reporting in two places.

That's said I think we can also merge it right now and probably clean up later - as my guess is that we need some cleanup in this area sooner or later anyway (and it is at least relatively clear to understand)

Reply:
Valid points. Our initial though was to gather metric information only about the connectors, and that might be easier to achieve with Grafana if we do not have all Job Worker's execution time. I think we should merge it know and we can clean this up later on. My suggestion would be to have the execution time for all Job Worker in the future. I created a follow up ticket: https://github.com/camunda/team-connectors/issues/304

Reply:
Make sense to me! For connectors having it for all Job workers would not change anything - as the connector runtime does not run any other job worker :-)